presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -226,6 +262,7 @@ private QueryResult setup(Query query, Function<String, QueryResult> executor)
     {
```

Comment:
I think you need to clear `preQueryResults` here, otherwise you'll need up with extra results in the code path where this is called from `checkForDeterministicAndRerunTestQueriesIfNeeded`